runtime/j9vm/javanextvmi.cpp

```diff
@@ -238,25 +250,158 @@ JVM_IsPreviewEnabled(void)
 JNIEXPORT void JNICALL
 JVM_VirtualThreadMountBegin(JNIEnv *env, jobject thread, jboolean firstMount)
 {
-	assert(!"JVM_VirtualThreadMountBegin unimplemented");
+	J9VMThread *currentThread = (J9VMThread *)env;
```

Comment:
Is it guaranteed that the passed in `thread` is always a virtual thread ?

Reply:
Yes. They are only used from the VirtualThread class: https://github.com/ibmruntimes/openj9-openjdk-jdk19/blob/034a8c8f1203b5b4e364d9a7dc4e750725c57634/src/java.base/share/native/libjava/VirtualThread.c#L42.

Reply:
As the caller is from another repo, you may consider adding an assertion that it is a virtual thread. Assert only in this function should be sufficient. (I am worried about the while loop a few lines below could become infinite if an incorrect obj is passed in from caller by mistake).

Reply:
Added the assertion in `JVM_VirtualThreadMountBegin`.